### PR TITLE
provisioning: Warns the user when uid or title is re-used.

### LINKED ISSUE #10880: Sanity checks and documentation for dashboard provisioning
Dashboards are currently being matched to existing dashboards in the database using `id`, `title` + `folder` and `uid`. This makes it so that dashboards as well as provisioning data might be overwritten over and over if multiples of these combinations exists. To handle this prior to a rewrite of the save logic we should add warnings and better documentation.

Warn when multiple dashboards are provisioned...
- [x] ...in the same folder with the same title
- [x] ...with the same UID

- [x] Update the documentation to warn about this behaviour

Related to #10865

### PR DESCRIPTION
**Pull request by xlson**
Closes #10880

### CODE REVIEW

File: pkg/services/provisioning/dashboards/file_reader.go [hunk 1]
```diff
@@ -280,3 +291,41 @@ func (fr *fileReader) readDashboardFromFile(path string, lastModified time.Time,
...
+func (checker provisioningSanityChecker) logWarnings(log log.Logger) {
+	for uid, times := range checker.uidUsage {
+		if times > 1 {
+			log.Error("the same 'uid' is used more than once", "uid", uid)
+		}
+	}
+
+	for title, times := range checker.titleUsage {
+		if times > 1 {
+			log.Error("the same 'title' is used more than once", "title", title)
```
Comment (roidelapluie) [inline pkg/services/provisioning/dashboards/file_reader.go]: is it per-folder?

Comment (xlson) [reply pkg/services/provisioning/dashboards/file_reader.go]: Good catch. I was incorrectly assuming folder to be specified in the provisioning config but that's not always the case. Thanks.

Comment (xlson) [reply pkg/services/provisioning/dashboards/file_reader.go]: It turns out that it actually isn't possible to set folderId in a provisioned dashboard, which makes sense as provisioning shouldn't depend on id's. It's still possible to have several provisioning configs using the same folder but solving that is out of scope of the current change.

**Review by bergquist — APPROVED** [review]

**Comment by bergquist** [comment]
Looks good to me!

**Pull request merged**